### ISSUE TITLE
ranking.listFeatures with streaming mode throws not implemented error and returns 0 hits

### ISSUE THREAD

**Issue by jobergum** [vespa-engine/vespa#29588]
Setting [ranking.listFeatures](https://docs.vespa.ai/en/reference/query-api-reference.html#ranking.listfeatures) to `true` with streaming mode causes the search to not return anything. 

```
schema mail {
    document mail {
        field features type tensor<float>(x{}) {
            indexing: summary | attribute
        }
    }
    rank-profile dot {
      inputs {
        query(q_features) tensor<float>(x{})
      }
      first-phase {
        expression: sum(query(q_features) * attribute(features))
      }
    }
}
```

Using listFeatures:

```
vespa query 'yql=select * from sources mail where true' 'streaming.groupname=x' 'ranking=default' 'ranking.listFeatures=true'
{
    "root": {
        "id": "toplevel",
        "relevance": 1.0,
        "fields": {
            "totalCount": 0
        },
        "coverage": {
            "coverage": 100,
            "documents": 1,
            "full": true,
            "nodes": 1,
            "results": 1,
            "resultsFull": 1
        }
    }
}
```

Error from vespa log 

```
[2023-12-08 07:32:59.166] WARNING searchnode       proton.visitor.instance.visitor	
	Visitor visitor-10-1702020779160-1-0-1652 failed in completedVisiting() callback.  As visitor is already complete, this has been ignored: UnsupportedOperationException: The function is not implemented for attribute 'features' of type 'search::tensor::TensorExtAttribute'.
	Backtrace:
	  /opt/vespa/lib64/libvespalib.so(vespalib::UnsupportedOperationException::UnsupportedOperationException(vespalib::stringref, vespalib::stringref, int)+0x18) [0xffff86b3c164]
	  /opt/vespa/lib64/libsearchlib.so(search::NotImplementedAttribute::notImplemented() const+0x88) [0xffff88f97420]
	  /opt/vespa/lib64/libsearchlib.so(+0x793adc) [0xffff8919eadc]
	  /opt/vespa/lib64/libsearchlib.so(search::features::AttributeMatchExecutor<search::attribute::AttributeContent<search::attribute::WeightedType<double> > >::Computer::run(unsigned int)+0x104) [0xffff8936ea84]
	  /opt/vespa/lib64/libsearchlib.so(search::features::AttributeMatchExecutor<search::attribute::AttributeContent<search::attribute::WeightedType<double> > >::execute(unsigned int)+0x30) [0xffff8936ed30]
	  /opt/vespa/lib64/libsearchlib.so(search::fef::Utils::extract_feature_values(search::fef::FeatureResolver const&, unsigned int, vespalib::FeatureSet::Value*)+0x13c) [0xffff8940c91c]
	  /opt/vespa/lib64/libstreamingvisitors.so(streaming::HitCollector::getFeatureSet(streaming::HitCollector::IRankProgram&, search::fef::FeatureResolver const&, vespalib::hash_map<vespalib::small_string<48u>, vespalib::small_string<48u>, vespalib::hash<vespalib::small_string<48u> >, std::equal_to<void>, vespalib::hashtable_base::and_modulator> const&)+0x108) [0xffff899c3d88]
	  /opt/vespa/lib64/libstreamingvisitors.so(streaming::RankProcessor::calculateFeatureSet()+0x98) [0xffff899cff5c]
	  /opt/vespa/lib64/libstreamingvisitors.so(streaming::SearchVisitor::RankController::onCompletedVisiting(vsm::GetDocsumsStateCallback&, vdslib::SearchResult&)+0x128) [0xffff899d8bb4]
	  /opt/vespa/lib64/libstreamingvisitors.so(streaming::SearchVisitor::completedVisitingInternal(storage::Visitor::HitCounter&)+0x70) [0xffff899dfd74]
	  /opt/vespa/lib64/libstreamingvisitors.so(streaming::SearchVisitor::completedVisiting(storage::Visitor::HitCounter&)+0x14) [0xffff899e0594]
	  /opt/vespa/lib64/libstorage.so(storage::Visitor::continueVisitor()+0x3ac) [0xffff882864cc]
	  /opt/vespa/lib64/libstorage.so(storage::VisitorThread::onInternalReply(std::shared_ptr<storage::api::InternalReply> const&)+0x128) [0xffff8829d808]
	  /opt/vespa/lib64/libstorage.so(+0x1c66dc) [0xffff882016dc]
	  /opt/vespa/lib64/libstorage.so(storage::VisitorThread::run(storage::framework::ThreadHandle&)+0x20c) [0xffff8829dbf4]
	  /opt/vespa/lib64/libstorage.so(storage::framework::defaultimplementation::ThreadImpl::run()+0x6c) [0xffff8838b2dc]
	  /lib64/libstdc++.so.6(+0xbcd2c) [0xffff7d5ddd2c]
	  /opt/vespa/lib64/vespa/malloc/libvespamalloc.so(+0x15c24) [0xffff89a94c24]
	  /lib64/libpthread.so.0(+0x78b8) [0xffff866aa8b8]
	  /lib64/libc.so.6(+0x23afc) [0xffff7d2dcafc]
```

**Comment by arnej27959** [comment]
this specific example should work in next release.

**Comment by geirst** [comment]
Fixed by https://github.com/vespa-engine/vespa/pull/30060. Available in Vespa 8.294.11.
